test/test_subparsers.py

```diff
@@ -0,0 +1,62 @@
+import argparse
+import contextlib
+import dataclasses
+import inspect
+import textwrap
+from dataclasses import dataclass, fields
+from enum import Enum
+from typing import *
+
+import pytest
+import simple_parsing
+import test
+from .testutils import TestSetup, xfail
+
+from simple_parsing.utils import subparsers
+
+
+@dataclass
+class TrainOptions:
+    """ Training Options """
+    lr: float = 1e-3
+    train_path: str = "train"
+
+@dataclass
+class ValidOptions:
+    """ Validation Options """
+    test_path: str = "test"
+    metric: str = "accuracy"
+
+@dataclass
+class GlobalOptions(TestSetup):
+    """ Global Options """
+    # mode, either Train or Valid.
+    mode: Union[TrainOptions, ValidOptions] = subparsers({
+        "train": TrainOptions,
+        "valid": ValidOptions,
+    }, default="train")
```

Comment:
Should people be forced to use either subcommand? or is something like "default" okay?